src/Raven.Server/Rachis/Leader.cs

```diff
@@ -557,18 +557,18 @@ public void Dispose()
                 _thread.Join();
 
             var ae = new ExceptionAggregator("Could not properly dispose Leader");
```

Comment:
This didn't actually fix the problem. We are still exposed to the modification during the `ToArray`